src/app/multi-line-chart/multi-line-chart.component.ts

```diff
@@ -268,13 +266,36 @@ export class MultiLineChartComponent implements OnInit {
   }
 
   private getCasesTicks(): number[] {
-    const [min, max] = this.yScale.domain();
-    const result = [];
-    for (let i = 1; i <= max; i *= 10) {
-      if (i >= min) {
-        result.push(i);
+    if (this.yAxisScale === AxisScale.log) {
+      const [min, max] = this.yScale.domain();
+      const result = [];
+      for (let i = 1; i <= max; i *= 10) {
+        if (i >= min) {
+          result.push(i);
+        }
       }
+      return result;
+    } else if (this.yAxisScale === AxisScale.linear) {
+      return this.yScale.ticks();
+    } else {
+      throw new UnreachableCaseError(this.yAxisScale);
     }
-    return result;
+  }
+
+  private getYScale(): d3.ScaleContinuousNumeric<number, number> {
+    let scale: d3.ScaleContinuousNumeric<number, number>;
+    if (this.yAxisScale === AxisScale.linear) {
+      scale = d3.scaleLinear();
+    } else if (this.yAxisScale === AxisScale.log) {
+      scale = d3.scaleLog();
+    } else {
+      throw new UnreachableCaseError(this.yAxisScale);
```

Comment:
I just learned about this the other day! If you don't know about it:

https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript